middlewares/metrics.go

```diff
@@ -34,9 +67,21 @@ func NewMetricsWrapper(impl Metrics) *MetricsWrapper {
 func (m *MetricsWrapper) ServeHTTP(rw http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
 	start := time.Now()
 	prw := &responseRecorder{rw, http.StatusOK}
+	fmt.Println(m.typ + " start")
 	next(prw, r)
-	labels := []string{"code", strconv.Itoa(prw.StatusCode()), "method", r.Method}
+	fmt.Println(m.typ + " stop")
+	labels := []string{"code", strconv.Itoa(prw.StatusCode()), "method", r.Method, "name", m.getName(r), "type", m.typ}
+
+	var state string
+	if prw.StatusCode() < 400 {
+		state = "Successful"
+	} else {
+		state = "Failing"
+	}
+
+	labelsStatus := []string{"state", state, "method", r.Method, "name", m.getName(r), "type", m.typ}
 	m.Impl.getReqsCounter().With(labels...).Add(1)
+	m.Impl.getReqsStatusCounter().With(labelsStatus...).Add(1)
```

Comment:
Sure, I was basing it on rancher-exporter that splits these metrics. But having them together is indeed better idea.